pkg/resources/object_renaming_acceptance_test.go

```diff
@@ -0,0 +1,1366 @@
+- are in config - schema renamed externally - with database implicit dependency - with implicit schema dependency (fails because tries to create database when it's already there after rename)
+- are in config - schema renamed externally - with database implicit dependency - with schema depends_on 		  (fails because tries to create database when it's already there after rename)
+
+- are in config - schema renamed externally - with no database dependency - with no schema dependency 		(fails because table is created before schema)
+- are in config - schema renamed externally - with no database dependency - with implicit schema dependency (fails because tries to create database when it's already there after rename)
+- are in config - schema renamed externally - with no database dependency - with schema depends_on 			(fails because tries to create database when it's already there after rename)
+
+- are in config - schema renamed externally - with database depends_on - with no schema dependency 		 (fails because table is created before schema)
+- are in config - schema renamed externally - with database depends_on - with implicit schema dependency (fails because tries to create database when it's already there after rename)
+- are in config - schema renamed externally - with database depends_on - with schema depends_on 		 (fails because tries to create database when it's already there after rename)
+
+------------------------------------------------------------------------------------------------------------------------
+
+- are not in config - database renamed externally - referencing old database name (fails because tries to create table on non-existing database)
+- are not in config - database renamed externally - referencing new database name (fails because tries to create table that already exists in the renamed database)
+
+- are not in config - schema renamed externally - referencing old schema name (fails because tries to create table on non-existing schema)
+- are not in config - schema renamed externally - referencing new schema name (fails because tries to create table that already exists in the renamed schema)
+
+# The list of test cases that were not added:
+- (Deep hierarchy) More test cases with varying dependencies between resources
+- (Deep hierarchy) Add test cases where old database is referenced to see if hierarchy recreation is possible
+- (Deep hierarchy) More test cases could be added when database and schema are renamed at the same time
+- (Deep hierarchy) More test cases could be added when either database or schema are in the config
+*/
+
+type DependencyType string
+
+const (
+	ImplicitDependency  DependencyType = "implicit"
+	DependsOnDependency DependencyType = "depends_on"
+	NoDependency        DependencyType = "no_dependency"
+)
+
+func TestAcc_ShallowHierarchy_IsInConfig_RenamedInternally_WithImplicitDependency(t *testing.T) {
```

Comment:
question: do you think that setting up parametrized tests for the shallow hierarchy too would also help in readability?

Reply:
Yes, after I wrote the deeper hierarchy I saw how the tests are much more compact (and readable). I wanted to rewrite shallow ones, but on the other hand, I thought to leave them as they are just because they're already there and maybe it's better to leave them in separation and avoid if-ology in test setups (if possible). Don't have strong feelings for this one 😅 wdyt?

Reply:
I have the same feeling that the deeper ones are much more readable now. Usually, I am against complex setup in parametrized tests but this use case is a bit different, we care about the permutations so setting the tests this way would improve readability and make them more understandable. I also don't feel a need for a strong push to rewrite them now, maybe in one of the next PRs or something

Reply:
keeping it open then